src/shimWeb3.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * If no existing window.web3 is found, this function injects a web3 "shim" to
+ * not break dapps that rely on window.web3.currentProvider.
+ *
+ * @param {import('./MetaMaskInpageProvider')} provider - The provider to set as window.web3.currentProvider.
+ */
+module.exports = function shimWeb3 (provider) {
+  if (!window.web3) {
+    const SHIM_IDENTIFIER = '__isMetaMaskShim__'
+    const web3Shim = new Proxy(
+      {
+        currentProvider: provider,
+        [SHIM_IDENTIFIER]: true,
+      },
+      {
+        get: (target, property, ...args) => {
+          if (property === 'currentProvider') {
+            console.warn(
+              'You are accessing the MetaMask window.web3.currentProvider shim. Just use window.ethereum instead. For details, see: https://docs.metamask.io/guide/provider-migration.html#replacing-window-web3',
```

Comment:
How does this sound?
```suggestion
              'You are accessing the MetaMask window.web3.currentProvider shim. This property is deprecated; use window.ethereum instead. For details, see: https://docs.metamask.io/guide/provider-migration.html#replacing-window-web3',
```

I have an aversion to using the word "just" in error messages because it can feel dismissive of people's confusion, and I think calling this "deprecated" better communicates the status of this property.